pkg/apis/kubernetes/v1/node_pool_types.go

```diff
@@ -0,0 +1,57 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/apis/internal"
+)
+
+// anxcloud:object:hooks=RequestBodyHook
+
+// NodePool represents a Kubernetes node pool
+// This resource does not support updates
+type NodePool struct {
+	internal.CommonMethods
+	State *internal.State `json:"state,omitempty"`
+
+	Identifier string  `json:"identifier,omitempty" anxcloud:"identifier"`
+	Name       string  `json:"name,omitempty"`
+	Cluster    Cluster `json:"cluster,omitempty"`
+
+	// Number of replicas can be changed via machine controller. Default: 3 (see FAQ for more details)
+	Replicas *int     `json:"replicas,omitempty"`
+	CPUs     int      `json:"cpus,omitempty"`
+	Memory   gibibyte `json:"memory,omitempty"`
+	DiskSize gibibyte `json:"disk_size,omitempty"`
+
+	OperatingSystem string `json:"operating_system,omitempty"`
+}
+
+const gibiFactor = 1073741824
+
+type gibibyte int
+
+func (v *gibibyte) UnmarshalJSON(b []byte) error {
+	var val int
+	if err := json.Unmarshal(b, &val); err != nil {
+		return fmt.Errorf("failed to unmarshal value: %w", err)
+	}
+	*v = gibibyte(val / gibiFactor)
+	return nil
+}
+
+func (v gibibyte) MarshalJSON() ([]byte, error) {
+	return json.Marshal(int(v) * gibiFactor)
+}
```

Comment:
do I read this correctly as "v is given in GiB, so we multiply with gibifactor to send Bytes to the API"? Why not using bytes in the first place then?